src/audio/host.c

```diff
@@ -264,10 +292,12 @@ static void host_dma_cb(void *arg, enum notify_id type, void *data)
 static uint32_t host_get_copy_bytes_one_shot(struct comp_dev *dev)
 {
 	struct host_data *hd = comp_get_drvdata(dev);
```

Comment:
There are a lot of ifdefs sprinkled all over the code which makes it very hard to read. I think would be better to implement a new version of host_get_copy_bytes_one_shot first guarded by the ifdef. 
Maybe letter we can unify them if we can do it in an elegant way.

Reply:
Agree. I'll refine it in another way.